simulator/src/main/java/com/hedera/block/simulator/mode/PublisherModeHandler.java

```diff
@@ -80,13 +80,17 @@ public void start() throws BlockSimulatorParsingException, IOException, Interrup
     }
 
     private void millisPerBlockStreaming() throws IOException, InterruptedException, BlockSimulatorParsingException {
-
         final long secondsPerBlockNanos = (long) millisecondsPerBlock * NANOS_PER_MILLI;
 
         Block nextBlock = blockStreamManager.getNextBlock();
         while (nextBlock != null) {
             long startTime = System.nanoTime();
-            publishStreamGrpcClient.streamBlock(nextBlock);
+            final boolean streamSuccess = publishStreamGrpcClient.streamBlock(nextBlock);
+            if (!streamSuccess) {
+                LOGGER.log(System.Logger.Level.INFO, "Block Stream Simulator stopped streaming due to errors.");
+                break;
+            }
```

Comment:
I think you can skip the variable assignment and just have:
```
if (!publishStreamGrpcClient.streamBlock(nextBlock)) {
   // log message
  break;
}
```

Reply:
Changed, although I usually like to assign, just for better readability.